communication/nats/dialog/dialog_waiter.go

```diff
@@ -53,17 +53,19 @@ type dialogWaiter struct {
 	sync.RWMutex
 }
 
+// Start registers dialogWaiter with broker (NATS) service
 func (waiter *dialogWaiter) Start() (dto_discovery.Contact, error) {
-	log.Info(waiterLogPrefix, fmt.Sprintf("Connecting to: %#v", waiter.myAddress))
+	log.Info(waiterLogPrefix, "Connecting to: ", waiter.myAddress.GetContact())
 
 	err := waiter.myAddress.Connect()
 	if err != nil {
-		return dto_discovery.Contact{}, fmt.Errorf("failed to start my connection. %s", waiter.myAddress)
+		return dto_discovery.Contact{}, log.Error(waiterLogPrefix, "failed to start my connection with: ", waiter.myAddress.GetContact())
```

Comment:
It suppose to return error object, not call log function

Reply:
We need informative error message, its printed out on startup log and log.Error() returns error.

Reply:
But we dont need to log here.
Just format and return error